src/core/unit.c

```diff
@@ -3456,6 +3456,7 @@ static int get_name_owner_handler(sd_bus_message *message, void *userdata, sd_bu
                         log_unit_error_errno(u, r,
                                              "Unexpected error response from GetNameOwner(): %s",
                                              bus_error_message(e, r));
+                        return 0;
```

Comment:
hmm, I am not sure if I am onboard with this. I think a bus service on a broker that doesn't work can be assumed to be dead. 

It appears to me as if this change makes things worse: we'll have a service we consider running even though we cannot talk to it.

Can you elaborate on the specific issue you saw here, and what kind of error you ran into specifically?